### PR TITLE
Reduce clippy warnings

### PR DESCRIPTION
**Pull request by Guiguiprim**
Contributing to #1874

I made one commit by lint rule, so we can easily drop any of them if you do not care for it.

I did not handle:
- `clippy::match_single_binding`: kind of false-positive due to temporary life prolongation hack (did not add a `allow`)
- `clippy::await_holding_refcell_ref`
    - in `compiler/passes.rs`: it could be fixed with some cloning but I hope the panic issue can not be triggered so we do not need to.
    - in `lsp` there is two occurrences on `document_cache` (in `main` and `server_loop`), I do not know the code enough to rule out a danger here

### CODE REVIEW

File: internal/core/api.rs [hunk 1]
```diff
@@ -40,11 +40,11 @@ impl LogicalPosition {
 
     /// Convert this logical position to a physical position by multiplying the coordinates with the
     /// specified scale factor.
-    pub fn to_physical(&self, scale_factor: f32) -> PhysicalPosition {
-        PhysicalPosition::from_logical(*self, scale_factor)
+    pub fn to_physical(self, scale_factor: f32) -> PhysicalPosition {
```
Comment (ogoffart) [inline internal/core/api.rs:43]: This, and the other in this file, are part of the public API.   I wonder if it's safe to do this change.

**Review by ogoffart — COMMENTED** [review]
Thanks,  

See my comment about possibly breaking change.

File: internal/core/api.rs (hunk 1 above)
Comment (Guiguiprim) [reply internal/core/api.rs:43]: Right. I think it's OK.

To be able to call `to_physical()`, users had to have reference on types that are and were copy.
So all call to `to_physical()` will still work as they will now copy if needed.

The following was working and still is.

```rust
let a = LogicalPosition { x: 1.0, y: 1.0 };
let x = a.to_physical(); // was taking &self, is now taking self takes to Copy
```

The only thing I see is that some `needless_borrow` warning could be added, but the chances are pretty low I think.

Comment (ogoffart) [reply internal/core/api.rs:43]: Unless someone did:

```rust
let a = LogicalPosition { x: 1.0, y: 1.0 };
LogicalPosition::to_physical(&a, 1.0); 
```

But perhaps nobody does that?

Comment (Guiguiprim) [reply internal/core/api.rs:43]: Rarely I think, and that is the case where they would get the clippy warning. It would still work.

Comment (Guiguiprim) [reply internal/core/api.rs:43]: Actually it does not work 😢

Comment (ogoffart) [reply internal/core/api.rs:43]: I believe it's better to #[allow] them for now.
We can open an issue about API changes we can do in Slint 2

Comment (Guiguiprim) [reply internal/core/api.rs]: Actually, when fixing this warning I when further that just what clippy was warning about.

To fix the warning I only need to change two `to_euclid()` which are `pub(crate)`  so this is not breaking.
I removed the other changes I made to align things that was not need and was breaking.

Comment (ogoffart) [reply internal/core/api.rs]: Filled https://github.com/slint-ui/slint/issues/3023

**Pull request merged**